c2corg_api/views/association.py

```diff
@@ -76,6 +78,9 @@ def collection_delete(self):
             if association is None:
                 raise HTTPBadRequest('association does not exist')
 
+        if is_main_waypoint_association(association):
+            raise HTTPBadRequest('waypoint is still main waypoint of route')
```

Comment:
Perhaps the error could be more explicit, like ``As the main waypoint of the route, this waypoint cannot be deassociated``?